query_optimizer/physical/HashJoin.hpp

```diff
@@ -50,9 +53,28 @@ typedef std::shared_ptr<const HashJoin> HashJoinPtr;
  */
 class HashJoin : public BinaryJoin {
  public:
+  enum JoinType {
+    kInnerJoin = 0,
+    kLeftSemiJoin,
+    kLeftAntiJoin
+  };
+
   PhysicalType getPhysicalType() const override { return PhysicalType::kHashJoin; }
 
-  std::string getName() const override { return "HashJoin"; }
+  std::string getName() const override {
+    switch (join_type_) {
+      case kInnerJoin:
+        return "HashJoin";
+      case kLeftSemiJoin:
+        return "HashLeftSemiJoin";
+      case kLeftAntiJoin:
+        return "HashLeftAntiJoin";
+      default:
+        LOG(FATAL) << "Invalid JoinType: "
+                   << static_cast<typename std::underlying_type<JoinType>::type>(join_type_);
+    }
+    QUICKSTEP_UNREACHABLE();
```

Comment:
I think we could remove this line, as we either return a value, or abort the program.